src/Symfony/Component/Yaml/Parser.php

```diff
@@ -171,6 +171,7 @@ public function parse($value, $flags = 0)
                     $this->refs[$isRef] = end($data);
                 }
             } elseif (
+                // this regexp is backtracking-heavy and makes the parser fail on long strings
                 self::preg_match('#^(?P<key>'.Inline::REGEX_QUOTED_STRING.'|(?:![^\s]+\s+)?[^ \'"\[\{!].*?) *\:(\s+(?P<value>.+?))?\s*$#u', $this->currentLine, $values)
```

Comment:
Changing the regex to `self::preg_match('#^(?P<key>'.Inline::REGEX_QUOTED_STRING.'|[^ \'"\[\{!].*?) *+\:(\s++(?P<value>.+?))?\s*+$#u', $this->currentLine, $values)` doesn't seem to be enough. 
Do you see another possible improvement?